### PR TITLE
hwmv2: Fix problems with common board functionality, alternative.

### PR DESCRIPTION
**Pull request by tejlmand**
Alternative to #70548

This PR accomplishes the same as #70548 but is less intrusive.

The issues: #70445 and #70283 was caused by merging of files, not because of shortened file names.

Instead of enforcing full normalized board target names for files, then this PR allows for shortened normalized board target names without the SoC name. This is allowed for boards which defines a single SoC.

Some file rename is still required, namely files which relied on the merge feature, but for files which solely used the merge feature to allow a shortened file name, then no changes are required.

This significantly reduces number of changes required to fix the referenced issues.

To minimize user mistakes, then a build error is raised in following situations:
- If both a full normalized board target and shortened board target name are used to name a file, for example:
   rpi_pico.overlay together with rpi_pico_rp2040.overlay
- If a shortened board target name are used when the board defines two or more SoCs, for example on the nrf52480dk (real: nrf52840, emulated: nrf52811):
   nrf52840.overlay, because this file is now ambiguous  because the nrf52840dk defines two SoCs.

### CODE REVIEW

**Review by erwango — DISMISSED** [review]
Great improvment compared to the previous proposal. Thanks

**Review by aescolar — COMMENTED** [review]
This seems quite nice to me. Just a request to not duplicate equal dts files, but to include from one the other

File: boards/native/native_sim/native_sim_64.dts [hunk 1]
```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (c) 2019 Jan Van Winkel (jan.van_winkel@dxplore.eu)
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <posix/posix.dtsi>
```
[the remark this answers is not in the document]
Comment (tejlmand) [reply boards/native/native_sim/native_sim_64.dts]: Done

**Review by erwango — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Review by aescolar — APPROVED** [review]
Ok for the overall idea and the changes to the native targets

**Review by fabiobaltieri — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Comment by nashif** [comment]
@evgeniy-paltsev  @ruuddw  please review

**Review by ruuddw — APPROVED** [review]

**Pull request merged**